infra/launch_on_ray.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/python
+# Similar to launch.py, but this instead launches on a Ray cluster configured with auto-scaling TPUs
```

Comment:
Is the eventual goal to deprecate launch.py (just to avoid maintaining too many things)?

Reply:
No, it's important to have this too b/c some of our users are too cheap to use a ray cluster, and I'll sometimes want to launch quick one-offs.